ovirt_imageio/client/_upload.py

```diff
@@ -0,0 +1,169 @@
+# ovirt-imageio
+# Copyright (C) 2022 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+"""
+Upload commands.
+"""
+
+import os
+from contextlib import closing
+from collections import namedtuple
+
+from . import _api
+from . import _options
+from . import _ovirt
+from . import _ui
+
+
+class DiskInfo(namedtuple(
+        "DiskInfo", "name,initial_size,provisioned_size,content_type,format")):
+    """
+    Container class for attributes that the uploaded disk will have.
+    """
+
+    def __new__(cls, name, initial_size,
+                provisioned_size, content_type, format):
+        return tuple.__new__(cls, (
+            name, initial_size, provisioned_size, content_type, format))
+
+
+def register(parser):
+    cmd = parser.add_sub_command(
+        "upload-disk",
+        help="Upload disk",
+        func=upload_disk)
+
+    cmd.add_argument(
+        "-s", "--storage-domain",
+        required=True,
+        help="Name of the storage domain.")
+
+    cmd.add_argument(
+        "-f", "--format",
+        choices=("raw", "qcow2"),
+        default="qcow2",
+        help="Upload image format (default qcow2).")
+
+    cmd.add_argument(
+        "--sparse",
+        action="store_true",
+        help="Create sparse disk. Cannot be used with raw format on "
+             "block storage.")
+
+    cmd.add_argument(
+        "--id", "--disk-id",
+        type=_options.UUID(),
+        help="A UUID for the new disk. If not specified oVirt will "
+             "create a new UUID.")
+
+    cmd.add_argument(
+        "--disable-backup",
+        dest="enable_backup",
+        action="store_false",
+        help="Creates a disk that can not be used for incremental backup.")
+
+    cmd.add_argument(
+        "filename",
+        type=_options.file,
+        help="Path to image (e.g. /path/to/image.raw). "
+             "Supported formats: raw, qcow2, iso.")
+
+
+def _create_disk_info(filename, dst_format):
+    # Obtain the image info dictionary.
+    img_info = _api.info(filename)
+    if img_info["format"] not in ("qcow2", "raw"):
```

Comment:
Is an iso-image also "raw"? Asking because above it is listed as a supported "format", which seems inconsistent.

Reply:
Yes, here format will contain "raw" for iso images, and the command will create the disk with both iso extension and content_type.